javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -0,0 +1,337 @@
+package java.util
+package concurrent
+
+import java.lang.{reflect => jlr}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+import scala.scalajs._
+
+class CopyOnWriteArrayList[E] private(private var inner: js.Array[E])
+    extends List[E] with RandomAccess with Cloneable with Serializable {
+  self =>
+
+  // requiresCopyOnWrite is false if and only if no other object
+  // (like the iterator) may have a reference to inner
+  private var requiresCopyOnWrite = false
+
+  def this() = {
+    this(new js.Array[E])
+  }
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  def this(toCopyIn: Array[E]) = {
+    this()
+    toCopyIn.foreach(add)
+  }
+
+  def size(): Int =
+    inner.size
+
+  def isEmpty: Boolean =
+    size == 0
+
+  def contains(o: scala.Any): Boolean =
+    iterator.exists(o === _)
+
+  def indexOf(o: scala.Any): Int =
+    indexOf(o.asInstanceOf[E], 0)
+
+  def indexOf(e: E, index: Int): Int = {
+    checkIndexInBounds(index)
+    index + listIterator(index).indexWhere(_ === e)
+  }
+
+  def lastIndexOf(o: scala.Any): Int =
+    lastIndexOf(o.asInstanceOf[E], 0)
+
+  def lastIndexOf(e: E, index: Int): Int = {
+    @tailrec
+    def findIndex(iter: ListIterator[E]): Int = {
+      if (!iter.hasPrevious) -1
+      else if (iter.previous() === e) iter.nextIndex
+      else findIndex(iter)
+    }
+    findIndex(listIterator(size))
+  }
+
+  override def clone(): AnyRef =
+    new CopyOnWriteArrayList[E](this)
+
+  def toArray(): Array[AnyRef] =
+    toArray(new Array[AnyRef](size))
+
+  def toArray[T](a: Array[T]): Array[T] = {
+    val toFill: Array[T] =
+      if (a.size >= size) a
+      else jlr.Array.newInstance(
+          a.getClass.getComponentType, size).asInstanceOf[Array[T]]
```

Comment:
The `else` branch must be on its line if it's too long. If it has more than one line, it needs `{}`. The then branch must do the same. And then the entire if/else needs `{}`.

Or, you decide to force it one line despite the long line, on the grounds that all these additional `{}` will make it much uglier.